examples/mysql/mysql.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"path/filepath"
+	"time"
+
+	"github.com/CyberOwlTeam/flyway"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/go-connections/nat"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/google/uuid"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/mysql"
+	"github.com/testcontainers/testcontainers-go/network"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	mysqlDBVersion  = "8.0.36"
+	mysqlPort       = "3306"
+	mysqlSrvName    = "mysql"
+	mysqlDBName     = "mysqldb"
+	mysqlDBUsername = "mysql-user"
+	mysqlDBPassword = "password"
+)
+
+// mysqlContainer represents an abstration of MySQLContainer.
+type mysqlContainer struct {
+	*mysql.MySQLContainer
+}
+
+func main() {
```

Comment:
interesting ... because I've never done this stuff before I'm a bit lost in the structure of this module .... do you NEED a main func in all modules ...?

Essentially as this is just an example, I would expect it to define "examples" or tests, but have no real code in it. Essentially you have copied a lot of code from the original fylway module into this, but it is a "program of sorts".

see ExampleRunContainer

Reply:
is the MySql testcontainer similar to the postgres container, except for mysql :)

Reply:
ahh yeah, I think I see it now ... because in the original example I had written a test and an example, that "extra code" was in the flyway_test.go file, here you've kinda created a runnable program rather than a test or example, I think fundamentally that is the only issue I can see